### PR TITLE
[1429] snc serializer part 5

### PR DESCRIPTION
**Pull request by defong**
### Context
search and compare serializer part 5
course variant interplay of `program_type`, `study_mode` & `qualification`

### Changes proposed in this pull request
Added serialization of related course variant touch points

### Guidance to review
Its going to get mess from henceforth, PR to make it better is welcome

part 5 ot x

Refer to the almost complete
https://github.com/DFE-Digital/manage-courses-backend/pull/377

For the full whack

FYI 

Part 1
https://github.com/DFE-Digital/manage-courses-backend/pull/381

Part 2
https://github.com/DFE-Digital/manage-courses-backend/pull/387

Part 3
https://github.com/DFE-Digital/manage-courses-backend/pull/390

Part 4
https://github.com/DFE-Digital/manage-courses-backend/pull/401

Probably easier to follow the commits

### Checklist

- [ ] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [ ] Tested by running locally

### CODE REVIEW

**Comment by defong** [comment]
[[L] Serialize course to sync with search-and-compare](https://trello.com/c/EVoUeZHg/1429-l-serialize-course-to-sync-with-search-and-compare)

File: app/serializers/search_and_compare/course_serializer.rb [hunk 1]
```diff
@@ -63,5 +63,48 @@ def get_subjects
...
+    attribute(:Route)                                 { get_route }
+    attribute(:IsSalaried)                            { is_salaried? }
+    attribute(:Mod)                                   { object.description }
+    attribute(:IncludesPgce)                          { get_include_pgce }
+
+    attribute(:FullTime)                              { object.part_time? ? 3 : 1 }
+    attribute(:PartTime)                              { object.full_time? ? 3 : 1 }
+
+    def get_route
+      route_names = {
```
Comment (benilovj) [inline app/serializers/search_and_compare/course_serializer.rb]: I think this should be something like `#program_type_description` on `Course`

Comment (defong) [reply app/serializers/search_and_compare/course_serializer.rb]: Tried and failed heres the extract of the two methods
```def program_type_description
    if school_direct_salaried_training_programme? then " with salary"
    elsif pg_teaching_apprenticeship? then " teaching apprenticeship"
    else ""
    end
  end

  def description
    study_mode_string = (full_time_or_part_time? ? ", " : " ") +
      study_mode_description
    qualifications_description + study_mode_string + program_type_description
  end```

Comment (defong) [reply app/serializers/search_and_compare/course_serializer.rb]: hmmm in fact not too quite sure what you on about, 

what i really want it to translate 
underscores and casing and somehow throw in the brackets
school_direct_salaried_training_programme: "School Direct (salaried) training programme"

**Pull request merged**